lib/rubygems/specification.rb

```diff
@@ -2713,6 +2713,11 @@ def validate packaging = true
       warning "#{executable_path} is missing #! line" unless shebang
     end
 
+    files.each do |file|
+      next unless File.symlink?(file)
+      warning "#{file} is a symlink, which is not supported on all platforms"
```

Comment:
I strongly approve!

I think [`test_validate_files`](https://github.com/rubygems/rubygems/blob/master/test/rubygems/test_gem_specification.rb#L2714] will cover both of these cases.